packages/server/src/utils.ts

```diff
@@ -183,21 +183,22 @@ function endResponse(serverResponse: NodeResponse) {
   serverResponse.end(null, null, null);
 }
 
-function getHeadersArray(headers: Headers) {
-  const headersArray: string[] = [];
+function getHeadersObject(headers: Headers) {
+  const headersObject: Record<string, string> = {};
   headers.forEach((value, key) => {
     if (key === 'set-cookie') {
       const setCookies = headers.getSetCookie?.();
       if (setCookies) {
         setCookies.forEach(setCookie => {
-          headersArray.push('set-cookie', setCookie);
+          headersObject['set-cookie'] = setCookie;
```

Comment:
I think this is missing a use case where there is multiple `set-cookies` headers isn't it ? If I well remember it's even why @ardatan made this code in the first place.

Reply:
oh you are right... 😅

Reply:
When you store the headers in an array I think it handles multiple set-cookie s as expected;
`['set-cookie', 'foo: bar', 'set-cookie', 'baz: quz', 'content-type', 'application/json']`
so it takes odd items as names and even ones as values.